drivers/interrupt_controller/intc_nxp_irqsteer.c

```diff
@@ -225,153 +225,331 @@
 	uint32_t regmap_size;
+	int irq_offset;
+	int irqs_num;
+	/* There is one output irq for each group of 64 inputs. */
+	int num_masters;
+	/* One register bit map can represent 32 input interrupts. */
+	int reg_num;
 	struct irqsteer_dispatcher *dispatchers;
 };
 
 struct irqsteer_dispatcher {
 	const struct device *dev;
-	/* which set of interrupts is the dispatcher in charge of? */
-	uint32_t master_index;
-	/* which interrupt line is the dispatcher tied to? */
-	uint32_t irq;
-	/* reference count for all IRQs aggregated by dispatcher */
+	int master_index;
+	int irq;
 	uint8_t irq_refcnt[CONFIG_MAX_IRQ_PER_AGGREGATOR];
 };
 
+/* Global variables */
 static struct k_spinlock irqstr_lock;
-
 static uint8_t l1_irq_refcnt[CONFIG_2ND_LVL_ISR_TBL_OFFSET];
-
 static struct irqsteer_dispatcher dispatchers[] = {
 	IRQSTEER_DECLARE_DISPATCHERS(DT_NODELABEL(irqsteer))
 };
 
-/* used to convert system INTID to zephyr INTID */
-static int to_zephyr_irq(uint32_t regmap, uint32_t irq,
-			 struct irqsteer_dispatcher *dispatcher)
+/* Native register operations (replacing HAL functions) */
+
+/**
+ * @brief Initialize IRQ_STEER hardware
+ */
+static void irqsteer_hw_init(const struct irqsteer_config *cfg)
+{
+	int reg_num = cfg->reg_num;
+	uintptr_t base = cfg->regmap_phys;
+	int reg_index = 0;
+
+	/* Disable all master interrupts */
+	IRQSTEER_WRITE32(base, CHAN_MINTDIS(reg_num), 0xFFFFFFFF);
+
+	/* Clear all channel masks */
+	for (; reg_index < reg_num; reg_index++) {
+		IRQSTEER_WRITE32(base, CHAN_MASK(reg_index, reg_num), 0);
+	}
+}
+
+static int irqsteer_get_reg_index(const struct irqsteer_config *cfg, int irq)
+{
+	return (cfg->reg_num - (irq - cfg->irq_offset) / IRQSTEER_INT_SRC_REG_WIDTH - 1);
+}
+
+/**
+ * @brief Get number of interrupts for a specific master
+ */
+static int irqsteer_get_master_irq_count(const struct irqsteer_config *cfg, int master_index)
+{
+	int count = IRQSTEER_AGGRAGATED_INT_NUM_PER_GRP;
+
+	/*
+	 * Each interrupt group(one register represents 32 input interrupts)
+	 * has 32 interrupt sources.
+	 *
+	 * There are two cases based on SOC integration:
+	 *
+	 * 1. The interrupt group count(number of CHn_MASKx registers) is
+	 *    even number.
+	 *    In this case, every two interrupt groups are connected to
+	 *    one interrupt master. So each master has 64 interrupt
+	 *    sources connected.
+	 *
+	 * 2. The interrupt group count(number of CHn_MASKx registers) is
+	 *    odd number.
+	 *    In this case, master 0 is connected to one interrupt group,
+	 *    while other masters are all conected to two interrupt groups.
+	 *    So master 0 has 32 interrupt sources connected, and
+	 *    other masters have 64 interrupt sources connected.
+	 */
+	if ((cfg->reg_num % 2) != 0)
+	{
+		if (master_index == 0)
+		{
+			count = IRQSTEER_INT_SRC_REG_WIDTH;
+		}
+	}
+
+	return count;
+}
+
+/**
+ * @brief Enable specific interrupt in IRQ_STEER
+ */
+static void irqsteer_enable_interrupt(const struct irqsteer_config *cfg, uint32_t irq)
```

Comment:
suggestion: combine `irqsteer_enable_interrupt` and `irqsteer_disable_interrupt` into a single function: `irqsteer_enable_disable_interrupt` (or something like that) that takes a bool as the third parameter: `enabled`.